File: 15 exercicios POO/5/golang/main.go
package main
import "fmt"

type Animal interface{
    EmitirSom() string
}

type Cachorro struct{
    nome string
}
func (c Cachorro) EmitirSom() string {
    return "AU AU"
}

type Gato struct {
    nome string
}
func (g Gato) EmitirSom() string {
    return "MIAU"
}

func emitirSons(animais []Animal){
    for _, animal := range animais {
        fmt.Println(animal.EmitirSom())
    }
}

func main() {
    cachorro := Cachorro{nome: "chorro"}

    gato := Gato{nome: "to"}

    animais := []Animal{cachorro, gato}

    emitirSons(animais)

}

File: 15 exercicios POO/5/java/src/Main.java
public class Main {

    public static void emitirSons(Animal[] animais) {
        for (Animal animal : animais) {
            System.out.println(animal.emitir_som());
        }
    }

    public static void main(String[] args) {
        Cachorro cachorro = new Cachorro("chorro");
        Gato gato = new Gato("to");

        Animal[] animais = {cachorro, gato};

        emitirSons(animais);
    }
}